client-datastore-cassandra/src/main/java/com/telecominfraproject/wlan/client/datastore/cassandra/ClientSessionDAO.java

```diff
@@ -392,21 +424,31 @@ public ClientSession deleteSession(int customerId, long equipmentId, MacAddress
                 queryArgs.addAll(equipmentIds);
             }
 
-            StringBuilder strb = new StringBuilder(100);
-            strb.append(" locationId in (");
-            for (int i = 0; i < locationIds.size(); i++) {
-                strb.append("?");
-                if (i < locationIds.size() - 1) {
-                    strb.append(",");
-                }
-            }
-            strb.append(") ");
-
-            query = strb.toString() + query;
+            query = " locationId in " + CassandraUtils.getBindPlaceholders(locationIds) + query;
             query_head = "select macAddress from client_session_by_location where ";
             
             filterOptions = FilterOptions.customer_and_location;
         }
+        
+        //add macSubstring filters
+        if (macSubstring != null) {
+        	queryArgs.clear();
+        	queryArgs.add("%" + macSubstring.toLowerCase() + "%");
+        	query_head = "select macAddress from client_session_by_mac where ";
+        	
+        	if (locationIds != null && !locationIds.isEmpty()) {
+        		queryArgs.addAll(locationIds);
+        		query = "and " + query;
+        	}
+        	if (equipmentIds != null && !equipmentIds.isEmpty()) {
+        		queryArgs.addAll(equipmentIds);
+            	query_head = "select macAddress from client_session_by_mac_and_equipment where ";
+        	}
+        	
+        	query = "macAddressString like ? " + query + " allow filtering";
+        	
+        	filterOptions = FilterOptions.customer_and_macAddress;
```

Comment:
This query will return client sessions for ALL the customers, not just for the mandatory customerId that is passed in. This needs to be re-worked.